docs/README.md

```diff
@@ -0,0 +1,368 @@
+
+# Overview
+The SDK proxy server is responsible for wrapping the SDK under test with an HTTP interface that can be used to call any of its public-facing methods with an arbitrary set of arguments,
+and returning the response to the caller (which will be the harness server). It will also allow configuration options to be passed which override the base API URL the SDK makes requests to.
+Setting that URL to the test harness server’s address allows the harness to mock the backend APIs to return data back (or to trigger errors).
+
+### About `entityType`
+To check that the SDK methods are returning a correct type of an instance, they will all contain an `entityType` in their response. `enityType` can be one of four values:
+  `Variable`, `User`, `Feature`, `Object`, `Void`
+
+<h1 id="devcycle-test-harness-proxy-client">Client</h1>
+
+SDK Client
+
+<a id="opIdcreateClient"></a>
+
+`POST /client`
+
+*Create a DVC client instance*
+
+> Body parameter
+
+```json
+{
+  "clientId": "string",
+  "sdkKey": "string",
+  "options": {},
+  "enableCloudBucketing": true
+}
+```
+
+<h3 id="createclient-parameters">Parameters</h3>
+
+|Name|In|Type|Required|Description|
+|---|---|---|---|---|
+|`clientId`|body|string|true|A unique ID that will be used to reference the client once it has been created|
+|`sdkKey`|body|string|false|A unique key used to initialize the DVC SDK|
+|`options`|body|object|false|An object of launch parameters. See [DevCycle Docs](https://docs.devcycle.com/docs/sdk/server-side-sdks/node#initialization-options) for a full list of options|
+|`enableCloudBucketing`|body|boolean|false|An option to use the cloud bucketing version of the SDK|
+
+> Example responses
+
+> Created
+
+```json
+{
+  "message": "Success"
+}
+```
+
+```json
+{
+  "exception": "Missing environment key! Call initialize with a valid environment key"
+}
+```
+
+> 400 Response
+
+```json
+{
+  "error": "Invalid request: missing clientId"
+}
+```
+
+<h3 id="createclient-responses">Responses</h3>
+
+|Status|Meaning|Description|
+|---|---|---|
+|201|[Created](https://tools.ietf.org/html/rfc7231#section-6.3.2)|Created|
+|400|[Bad Request](https://tools.ietf.org/html/rfc7231#section-6.5.1)|Bad Request|
+
+<h3 id="createclient-responseschema">Response Schema</h3>
+
+Status Code **201**
+
+|Name|Type|Required|Description|
+|---|---|---|---|
+|`» message`|string|false|A message indicating the result of initializing an SDK client|
+|`» exception`|string|false|An error from the SDK indicating an exception was thrown during initialization|
+
+Status Code **400**
+
+|Name|Type|Required|Description|
+|---|---|---|---|
+|`» error`|string|true|An error message indicating what went wrong with the request|
+
+### Response Headers
+
+|Status|Header|Type|Format|Description|
+|---|---|---|---|---|
+|201|Location|string||Location of the Client instance on the proxy server|
+
+<a id="opIdcommand"></a>
+
+`POST /client/{clientId}/command`
```

Comment:
this endpoint appears to be in the client section, but technically its a general endpoint for calling commands on anything based on its location header (that way you can also call methods on variables etc.)

The actual path for this would be something like 
`/:location/command`

Reply:
yeah, the only reason I decided not to do that is cause I thought that'd be too generalized. I thought it would make more sense to have individual 'command' sections for other instances that you can also call methods on. Personally I think it's better to be more specific when writing docs but I can generalize it and make it `/:location/command`, it would save us time later down the line since we wouldn't need to update the spec

Reply:
the reason it's important to say its for any "location" is that the design is specifically made so that anything which has been identified with a location header can have commands called on it. The spec doesn't actually define what the "location" format is supposed to be since its the concern of the proxy server how to describe where something is basically, and the harness is supposed to just use whatever the proxy says the location is.

Reply:
yup, makes sense